docs/config_entries_config_flow_handler.md

```diff
@@ -68,7 +68,26 @@ There are a few step names reserved for system use:
 
 ## Unique IDs
 
-A config flow can attach a unique ID, which must be a string, to a config flow to avoid the same device being set up twice. When a unique ID is set, it will immediately abort if another flow is in progress for this unique ID. You can also quickly abort if there is already an existing config entry for this ID. Config entries will get the unique ID of the flow that creates them.
+A config flow can attach a unique ID, which must be a string, to a config flow to avoid the same device being set up twice.
+
+By setting a unique ID, users will have the option to ignore the discovery of your config entry. That way, they won't be bothered about it anymore.
+If the integration uses Bluetooth, DHCP, HomeKit, Zeroconf/mDNS, USB, or SSDP/uPnP to be discovered, supplying a unique ID is required.
+
+If a unique ID isn't available, alternatively, the `bluetooth`, `dhcp`, `zeroconf`, `hassio`, `homekit`, `ssdp`, `usb`, and `discovery` steps can be omitted, even if they are configured in
+the integration manifest. In that case, the `user` step will be called when the item is discovered.
+
+Alternatively, if an integration can't get a unique ID all the time (e.g., multiple devices, some have one, some don't), a helper is available
+that still allows for discovery, as long as there aren't any instances of the integrations configured yet.
+
+```python
+if device_unique_id:
+    await self.async_set_unique_id(device_unique_id)
+await self._async_handle_discovery_without_unique_id()
```

Comment:
What is this an example of? It doesn't make sense to me. If we have a unique id, calling a function for the case where we don't have a unique id is weird.

Reply:
I'm not sure. I didn't add this example, I just moved it.
I can open a separate PR which improves the example, and then rebase this PR once that's approved and merged?

Reply:
Ok